tests/prague/eip3074_auth_authcall/test_auth.py

```diff
@@ -0,0 +1,212 @@
+"""
+Tests for EIP-3074's `AUTH` Instruction
+"""
+
+from typing import Tuple
+
+import pytest
+from coincurve.keys import PrivateKey, PublicKey
+from ethereum.base_types import U256
+from ethereum.crypto.hash import keccak256
+
+from ethereum_test_tools import (
+    Account,
+    Address,
+    Environment,
+    StateTestFiller,
+    TestAddress,
+    TestAddress2,
+    TestPrivateKey,
+    TestPrivateKey2,
+    Transaction,
+)
+from ethereum_test_tools.common.base_types import Hash
+from ethereum_test_tools.vm.opcode import Opcodes as Op
+
+REFERENCE_SPEC_GIT_PATH = "EIPS/eip-3074.md"
+REFERENCE_SPEC_VERSION = "3b5fcad6b35782f8aaeba7d4ac26004e8fbd720f"
+
+MAGIC = b"\x04"
+
+
+@pytest.mark.valid_from("Prague")
+def test_all_zeros(state_test: StateTestFiller) -> None:
+    """
+    Test `AUTH` opcode with all zero arguments.
+    """
+    env = Environment()
+    to = Address(0x100)
+    pre = {
+        to: Account(code=Op.SSTORE(1, Op.ISZERO(Op.AUTH(0, 0, 0)))),
+        TestAddress: Account(balance=1000000000000000000000),
+    }
+    tx = Transaction(
+        to=to,
+        gas_limit=50000,
+    )
+    post = {
+        to: Account(
+            code=b"\x60\x00\x60\x00\x60\x00\xf6\x15\x60\x01\x55", storage={"0x01": "0x01"}
+        ),
+    }
+
+    state_test(env=env, pre=pre, post=post, tx=tx)
+
+
+@pytest.mark.valid_from("Prague")
+def test_expand_memory(state_test: StateTestFiller) -> None:
+    """
+    Test an `AUTH` that expands memory.
+    """
+    env = Environment()
+    to = Address(0x100)
+    pre = {
+        to: Account(code=Op.SSTORE(1, Op.ISZERO(Op.AUTH(0, 24000, 1)))),
+        TestAddress: Account(balance=1000000000000000000000),
+    }
+    tx = Transaction(
+        to=to,
+        gas_limit=60000,
+    )
+    post = {
+        to: Account(storage={"0x01": "0x01"}),
+    }
+
+    state_test(env=env, pre=pre, post=post, tx=tx)
+
+
+def sign_authorization(
+    private_key: str, magic: bytes, chain_id: int, nonce: int, invoker: Address, commit: Hash
+) -> Tuple[Address, bytes]:
+    """
+    Sign an authorization and return the authority and calldata in the format
+    expected for `auth`.
+    """
+    signing_bytes = (
+        magic
+        + U256(chain_id).to_be_bytes32()
+        + U256(nonce).to_be_bytes32()
+        + invoker.rjust(32, b"\x00")
+        + commit
+    )
+
+    signing_hash = keccak256(signing_bytes)
+    signature_bytes = PrivateKey(secret=Hash(private_key)).sign_recoverable(
+        signing_hash, hasher=None
+    )
+    public_key = PublicKey.from_signature_and_message(signature_bytes, signing_hash, hasher=None)
+
+    authority = keccak256(public_key.format(compressed=False)[1:])[32 - 20 :]
+
+    r = signature_bytes[:32]
+    s = signature_bytes[32:64]
+    y_parity = signature_bytes[64:65].rjust(32, b"\x00")
+
+    return (Address(authority), y_parity + r + s)
```

Comment:
```suggestion
    return (Address(authority), y_parity + r + s + commit)
```
The commit also goes back into the calldata.